horace_core/sqw/@SQWDnDBase/private/smooth_dnd.m

```diff
@@ -0,0 +1,157 @@
+function dout = smooth_dnd(din, xunit, varargin)
+% Smooths a 1,2,3 or 4 dimensional dataset
+%
+%Syntax:
+%   >> dout = smooth_dnd (din, xunit, width, shape)
+%
+% Input:
+% ------
+%   din     Input dataset structure (data_sqw_dnd)
+%   xunit   If true, unit of measure is length of a unit along each axis
+%           If false, unit of measure is pixels
+%   width   Vector that sets the extent of the smoothing along each dimension.
+%          The interpretation of width depends on the argument 'shape' described
+%          below.
+%           If width is scalar, then the value is applied to all dimensions
```

Comment:
This definition should match the example: so `if width is a vector of size equal to the number of dimensions, then width(i) will be applied to the i'th dimension. Vector sizes different from the dimensionality are invalid.`